### ISSUE TITLE
[Bug Report]: Build amd library not valid

### ISSUE THREAD

**Issue by shaodahong** [web-infra-dev/rspack#2278]
### System Info

  System:
    OS: Linux 5.15 Debian GNU/Linux 11 (bullseye) 11 (bullseye)
    CPU: (2) x64 AMD EPYC
    Memory: 623.82 MB / 1.63 GB
    Container: Yes
    Shell: 5.1.4 - /bin/bash
  Binaries:
    Node: 16.17.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 8.15.0 - /usr/local/bin/npm
    Watchman: 20220528.183901.0 - /usr/local/bin/watchman
  npmPackages:
    @rspack/cli: ^0.1.0 => 0.1.0 
    @rspack/core: ^0.1.0 => 0.1.0 

### Details

when setting the library config and use rspack api run build, output file is not amd module
```
library: {
    name: "test",
    type: "amd",
  },
```

### Reproduce link

https://codesandbox.io/p/sandbox/snowy-cdn-cgr6t5?file=%2Fbuild.js&selection=%5B%7B%22endColumn%22%3A13%2C%22endLineNumber%22%3A9%2C%22startColumn%22%3A13%2C%22startLineNumber%22%3A9%7D%5D

### Reproduce Steps

1. `yarn`
2. `node build.js`

view `dist/TestA` or `dist/TestB` file

**Comment by underfin** [comment]
Closed at https://github.com/web-infra-dev/rspack/pull/2240, you can 0.1.1 version.

**Comment by shaodahong** [comment]
> Closed at #2240, you can 0.1.1 version.

@underfin Thanks, and i upgrade 0.1.1 version, build it, output file its weird
```
define('name', [deps], () => {
  var __webpack_exports__ = __webpack_require__(namePath)
})
```

if i want get `name` module, the module need return or use some var export like
```
define('name', [deps], () => {
  var __webpack_exports__ = __webpack_require__(namePath)
 return __webpack_exports__
})
```
